float32[] commands
float64 timestamp
string sender
---
bool result